gateway/src/apicast/policy/nginx_metrics/nginx_metrics.lua

```diff
@@ -118,19 +126,22 @@ function _M:metrics()
   end
 end
 
-local function report_req_response_time()
+local function report_req_response_time(service)
   -- Use ngx.var.original_request_time instead of ngx.var.request_time so
   -- the time spent in the post_action phase is not taken into account.
   local resp_time = tonumber(ngx.var.original_request_time)
-
   if resp_time and response_times then
-    response_times:observe(resp_time)
+    response_times:observe(resp_time, { service })
   end
 end
 
-function _M.log()
-  upstream_metrics.report(ngx.var.upstream_status, ngx.var.upstream_response_time)
-  report_req_response_time()
+function _M.log(_, context)
+  local service = context.service.id
+  if not service or not extended_metrics then
+    service = "all"
+  end
+  upstream_metrics.report(ngx.var.upstream_status, ngx.var.upstream_response_time, service)
+  report_req_response_time(service)
```

Comment:
I know this is unrelated to this PR. But we probably should also introduce a counter of how many requests the gateway processed (and have a label for each service). So prometheus can detect requests per minute for example.